utils/src/main/scala/za/co/absa/enceladus/utils/validation/field/FractionalFieldValidator.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.validation.field
+import za.co.absa.enceladus.utils.schema.MetadataKeys
+import za.co.absa.enceladus.utils.types.TypedStructField
+import za.co.absa.enceladus.utils.validation.{ValidationError, ValidationIssue}
+
+object FractionalFieldValidator extends FractionalFieldValidator
+
+class FractionalFieldValidator extends ScalarFieldValidator {
```

Comment:
```suggestion
object FractionalFieldValidator extends ScalarFieldValidator {
```
Can't this simply be the way you had it `ScalarFieldValidator`? I don't understand the pattern with the companion object extending it's own class. This might be a topic for CQC.

Reply:
Here it definitely can. Not expected to be inherited. (Unlike `ScalarFieldValidator` where I had to convert it to a class and add the object exactly for this reason)